apps/src/javalab/JavabuilderConnection.js

```diff
@@ -311,12 +328,33 @@ export default class JavabuilderConnection {
   }
 
   onClose(event) {
-    if (event.wasClean) {
+    // Event code 1000 is "connection closed normally", so we should treat
+    // it as an expected close event. For some reason many close events with code
+    // 1000 are not marked as clean. We should treat them as clean.
+    if (event.code === 1000 || event.wasClean) {
```

Comment:
Should we made a constant for this value?

Reply:
will do!